test: fix tls-inception

santigimeno opened the pull request:
Make sure all the data is read before checking its validity.
Remove `gotHello` variable and just check that the ssl `end` event
is received.

/cc @indutny

mscdex commented on test/parallel/test-tls-inception.js:
```diff
@@ -13,7 +13,6 @@ var path = require('path');
 var net = require('net');
 
 var options, a, b, portA, portB;
```
FWIW `portA` and `portB` can be removed too, they're not used.

mscdex commented:
LGTM

indutny commented:
One nit from @mscdex , otherwise LGTM. Thanks!

santigimeno commented:
Updated the PR removing the unused variables.

I have also added a fix for the problems reported here https://github.com/nodejs/node/issues/1012#issuecomment-151266824. I have sent this in a separate commit. Is this correct? or should I squash both commits into a single one? or a different PR?

/cc @indutny @mscdex

indutny commented on test/parallel/test-tls-inception.js:
```diff
@@ -12,10 +12,9 @@ var fs = require('fs');
+var options, a, b;
 
-var body = new Buffer(4000).fill('A');
+var body = new Buffer(400000).fill('A');
```
I think this is a bit over the top, probably? What do you think about a compromise with a `40000`?

santigimeno replied:
I've used that value because to reproduce the `ECONNRESET` error in `OS X` consistently.

indutny commented on test/parallel/test-tls-inception.js:
```diff
@@ -33,7 +32,7 @@ a = tls.createServer(options, function(socket) {
   socket.pipe(dest);
 
-  dest.on('close', function() {
+  dest.on('end', function() {
```
Reason?

santigimeno replied:
I've tried to explain it the best I can (from my understanding) in the commit log: https://github.com/santigimeno/node/commit/432a86eb88f69e8d94253087bfd707cfce73828c
```
When sending a very large buffer (400000 bytes) the test fails due to
the client socket from the `a` server erroring with `ECONNRESET`.
There's a race condition between the closing of this socket and the `ssl`
socket closing on the other side of the connection. To improve things,
destroy the socket as soon as possible: in the `end` event of the `dest`
socket.
```
Running the test with `dtruss` I could see the `ssl` closing and the `socket` sending data afterwards. This data was from the `SSL_shutdown` called when the `dest` socket was closed (I guess because both sockets were piped)
What I don't really understand is how `ssl` receives an `EOF`, thus emitting the `end` event, before the other end has sent the shutdown. (I'm not sure if it makes sense).

indutny replied:
There are two ways to EOF for TLS connection: TCP FIN packet, and TLS close_notify alert (i.o.w. shutdown).

santigimeno replied:
I see. Thanks for the explanation!

indutny commented:
LGTM, thank you!

indutny commented:
CI: https://ci.nodejs.org/job/node-test-pull-request/972/

indutny commented:
Unrelated test failures, landing!

indutny commented:
Landed in 86a3bd09b0bca3441ad720286f738957102d9343 and 3b94991bda0922af09b00c2338f364d901678aad. Thank you!

santigimeno commented:
Thanks
